primitives/src/lib.rs

```diff
@@ -28,19 +28,29 @@ extern crate std;
 extern crate actual_serde as serde;
 
 #[cfg(feature = "alloc")]
-pub mod locktime;
+pub mod locktime; // Maintains units module structure and re-exports everything.
 pub mod opcodes;
 pub mod sequence;
 
-#[rustfmt::skip]                // Keep public re-exports separate.
 #[doc(inline)]
 #[cfg(feature = "alloc")]
-pub use self::{
-    locktime::{absolute, relative},
-};
+pub use self::locktime::{absolute, relative};
 #[doc(inline)]
 pub use self::sequence::Sequence;
 
+// Mirror the module structure and re-exports of `units`.
+#[rustfmt::skip]                // Keep public unit re-exports separate.
+#[doc(inline)]
+pub use units::amount::{self, Amount, SignedAmount};
+#[cfg(feature = "alloc")]
+#[doc(inline)]
+pub use units::{
+    block::{self, BlockHeight, BlockInterval},
+    fee_rate::{self, FeeRate},
+    parse,
+    weight::{self, Weight},
+};
```

Comment:
In f1e84487cbcdbfca45146a1d285bb5f4b44373a7:

I guess this explicit reexport is to avoid conflicts with `serde`?  Can't you just replace the above `extern crate actual_serde as serde;` with `use units::serde`? Alternately I think it'd be fine to drop the reexport of `serde` from `units` and make every crate do their own import.

It's confusing that the comment says "mirror the module structure and re-exports of `units`" but then you are clearly flattening everything. It would make more sense to write `pub use units::{block, BlockHeight, BlockInterval, ...}` rather than reexporting stuff from submodules of `units` in the root of `primitives`.

I'd further prefer if this could just be `pub use units::*`.

Reply:
Moved discussion to #3037

Reply:
Lq9000905@gmail.com